withthegrid/node-rdkafka
# WIP: Feat consume a specific partition

Fixes: https://github.com/Blizzard/node-rdkafka/issues/1063

How to test:

- clone branch locally
- init git submodules with: `git submodule update --init --recursive`
- run `npm install`
- run kafka with a broker on localhost:9093
- run `node ./test-create-topic-with-messages.js` to create a topic with 2 partitions with each 500 messages in it. Verify via Redpanda that the topic has been created. Kill the command manually.
- run `node ./test-consume.js` to start the consumer. This should consume the 2 partitions at a different speed.

When testing multiple times, make sure to update the group.id so that the consumer group offset is reset.

Limitations:
This setup only works if you keep calling `.consume`. Flowing mode is not possible.. yet?

Todo before merge:
- [ ] Remove test scripts
- [ ] Add error handling / baton class to Queue

Todo before sending a PR to original repository
